### PR TITLE
restructured module, cleanup

### PR DESCRIPTION
**Pull request by mbuchetics**

### CODE REVIEW

File: Sources/GDPR/GDPRManager.swift [hunk 1]
```diff
@@ -73,16 +87,20 @@ public class GDPRManager {
 
     public func setServices(services: [Service]) {
```
Comment (wieweb) [inline Sources/GDPR/GDPRManager.swift:88]: `public func setServices(services: Set<Service>) {`
If you use a set you don't need to assert for duplicates.

File: Sources/GDPR/ViewModels/GDPRService.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+public struct GDPRService {
+    var id: String
+    var name: String
+    var description: String
+    var supportDeletion: Bool?
```
Comment (wieweb) [inline Sources/GDPR/ViewModels/GDPRService.swift:7]: `var supportDeletion: Bool`
Optional is not needed. Will be initiated with default value `false`

File: Sources/GDPR/ViewModels/Status.swift [hunk 3]
```diff
@@ -18,35 +10,35 @@ enum TermState: Codable {
...
-    
+
     init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: CodingKeys.self)
         if let value = try? values.decode(Date.self, forKey: .accepted) {
             self = .accepted(at: value)
-        } else if  (try? values.decode(String.self, forKey: .rejected)) != nil {
+        } else if (try? values.decode(String.self, forKey: .rejected)) != nil {
             self = .rejected
         } else if (try? values.decode(String.self, forKey: .undefined)) != nil {
```
Comment (wieweb) [inline Sources/GDPR/ViewModels/Status.swift:26]: `} else if (try? values.decode(String.self, forKey: .undefined)) != nil {`
This case is not needed. Will be set to `.undefined` anyway.

**Review by wieweb — APPROVED** [review]

File: Sources/GDPR/GDPRManager.swift (hunk 1 above)
Comment (mbuchetics) [reply Sources/GDPR/GDPRManager.swift:88]: but we want to control the order of the services

**Review by draskovits — APPROVED** [review]

**Pull request merged**